### ISSUE TITLE
Massive refactor for clean up required

### ISSUE THREAD

**Issue by bashleigh** [bashleigh/typeorm-polymorphic#2]
I built this in a rush, the original I wrote on a plane, this version was written on a train. Yea it wasn't that extravigant. But the codebase is seriously poor. Needs more than prettier. Could use some help but it is a bit... ...my shitty mess :joy:

**Comment by codeninja** [comment]
Hush, it's fantastic and it solves my problem (at least in the 2.8 version of TypeORM). I would be happy to colab with you to bring the code up to the current 3.1 version which contains breaking changes.

**Comment by bashleigh** [comment]
Yes please! I also need to make it a lot more organised so that I can also fix an issue with parent/child saves (forget which)
